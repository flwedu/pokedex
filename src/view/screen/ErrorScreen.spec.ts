import {
  BadRequestError,
  InternalServerError,
  NotFoundError,
} from "../../util/custom-errors";
import { ErrorScreen } from "./";

describe("Error Screen tests", () => {
  it.each([
    new BadRequestError(""),
    new NotFoundError(""),
    new InternalServerError(""),
    new Error(),
  ])("should render the error screen", (error) => {
    const screen = new ErrorScreen();
    const html = screen.getHtml(error);

    expect(html).toMatchSnapshot();
  });
});
