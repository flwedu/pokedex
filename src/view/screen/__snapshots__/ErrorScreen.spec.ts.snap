// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Screen tests should render the error screen 1`] = `
"<div>
  <h2>BadRequestError</h2>
  <p></p>
  <img src=\\"../../../assets/error.svg\\" alt=\\"X\\">
  </div>"
`;

exports[`Error Screen tests should render the error screen 2`] = `
"<div>
  <h2>NotFoundError</h2>
  <p></p>
  <img src=\\"../../../assets/error.svg\\" alt=\\"X\\">
  </div>"
`;

exports[`Error Screen tests should render the error screen 3`] = `
"<div>
  <h2>InternalServerError</h2>
  <p></p>
  <img src=\\"../../../assets/error.svg\\" alt=\\"X\\">
  </div>"
`;

exports[`Error Screen tests should render the error screen 4`] = `
"<div>
  <h2>Error</h2>
  <p></p>
  <img src=\\"../../../assets/error.svg\\" alt=\\"X\\">
  </div>"
`;
